flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InMemoryReporter.java

```diff
@@ -131,12 +140,12 @@ void applyRemovals() {
     }
 
     private Stream<Entry<String, Metric>> getMetricStream() {
-        return metrics.entrySet().stream().flatMap(this::getGroupMetricStream);
+        return getMetrics().entrySet().stream().flatMap(this::getGroupMetricStream);
```

Comment:
I think this is not how you make this class thread safe. After `getMetrics()` we have access to `metrics` that can be changed by a different call.

Reply:
Yes, you are right. I'll fix it.